provider/kubernetes/kubernetes.go

```diff
@@ -421,9 +435,45 @@ func getRuleForPath(pa extensionsv1beta1.HTTPIngressPath, i *extensionsv1beta1.I
 		}
 		rules = append(rules, ruleTypeReplacePath+":"+rootPath)
 	}
+
+	if requestModifier := getStringValue(i.Annotations, annotationKubernetesRequestModifier, ""); requestModifier != "" {
+		rule, err := parseRequestModifier(requestModifier, ruleType)
+		if err != nil {
+			return "", err
+		}
+
+		rules = append(rules, rule)
+	}
 	return strings.Join(rules, ";"), nil
 }
 
+func parseRequestModifier(requestModifier, ruleType string) (string, error) {
+	if strings.Trim(requestModifier, " :") == "" {
```

Comment:
Extracting the `Trim` call into a variable allows reusing in line 456.

Additionally, we don't log any blanks-only value, thus allowing for a clearer error message.